WordPress/src/main/java/org/wordpress/android/ui/reader/models/ReaderRelatedPostList.java

```diff
@@ -2,17 +2,18 @@
 
 import android.support.annotation.NonNull;
 
-import org.wordpress.android.models.ReaderPost;
-import org.wordpress.android.models.ReaderPostList;
+import org.json.JSONArray;
 
 import java.util.ArrayList;
 
 public class ReaderRelatedPostList extends ArrayList<ReaderRelatedPost> {
 
-    public ReaderRelatedPostList(@NonNull ReaderPostList posts) {
-        for (ReaderPost post: posts) {
-            add(new ReaderRelatedPost(post));
+    public static ReaderRelatedPostList fromJsonPosts(@NonNull JSONArray jsonPosts) {
+        ReaderRelatedPostList posts = new ReaderRelatedPostList();
+        for (int i = 0; i < jsonPosts.length(); i++) {
+            posts.add(ReaderRelatedPost.fromJson(jsonPosts.optJSONObject(i)));
```

Comment:
I don't think it's possible for `optJSONObject()` to return null here, since we're looping through the array of objects. But for future use null-checking makes sense - added in 69c0909.